modules/library/jdbc/src/test/java/org/geotools/jdbc/JDBCAggregateFunctionOnlineTest.java

```diff
@@ -415,6 +415,24 @@ public void testUniqueWithLimitOnVisitor() throws Exception {
         assertEquals("one", result.iterator().next());
     }
 
+    @Test
+    public void testUniqueWithNonMatchingSort() throws Exception {
+        assumeTrue(dataStore.getSQLDialect().isAggregatedSortSupported("distinct"));
+
+        FilterFactory ff = dataStore.getFilterFactory();
+        PropertyName p = ff.property(aname("stringProperty"));
+
+        UniqueVisitor v = new MyUniqueVisitor(p);
+        v.setStartIndex(0);
+        Query q = new Query(tname("ft1"));
+        PropertyName pNonMatching = ff.property(aname("doubleProperty"));
+        q.setSortBy(new SortByImpl(pNonMatching, ASCENDING));
+        dataStore.getFeatureSource(tname("ft1")).accepts(q, v, null);
```

Comment:
```suggestion
        // used to fail with an exception, because stringProperty was in order by, but not in select
        dataStore.getFeatureSource(tname("ft1")).accepts(q, v, null);
```

Reply:
Fixed